Kernel/Storage/StorageManagement.cpp

```diff
@@ -241,8 +241,15 @@ MajorNumber StorageManagement::storage_type_major_number()
 }
 MinorNumber StorageManagement::generate_storage_minor_number()
 {
-    auto minor_number = s_device_minor_number.load();
-    s_device_minor_number++;
+    auto minor_number = s_storage_device_minor_number.load();
+    s_storage_device_minor_number++;
+    return minor_number;
+}
+
+MinorNumber StorageManagement::generate_partition_minor_number()
+{
+    auto minor_number = s_partition_device_minor_number.load();
+    s_partition_device_minor_number++;
```

Comment:
Same here, though I think this is less critical as within a device we'll probably always be initializing partitions in a simple loop.